hostfactory/1.1/providerplugins/azurecc/scripts/src/cyclecloud_provider.py

```diff
@@ -662,14 +670,15 @@ def _create_status(self, input_json, json_writer=None):
                 report_failure_states = ["Unavailable"]
                 terminate_states = ["Failed"]
             
-            for node in requested_nodes["nodes"]:
+            for node in requested_nodes:
                 # for new nodes, completion is Ready. For "released" nodes, as long as
                 # the node has begun terminated etc, we can just say success.
                 # node_status = node.get("State")
-                node_status = node.get("Status")
-
-                node_target_state = node.get("TargetState", "Started")
-                all_nodes.append(node["NodeId"])
+                if isinstance(node, str):
```

Comment:
When can this happen?   (seems like we've got something wrong somewhere else if we can get a str here)